core/src/odbc_uri.rs

```diff
@@ -250,9 +251,15 @@ impl<'a> ODBCUri<'a> {
             );
         }
         Self::set_server_and_source(&mut client_options, server, source)?;
+        client_options.app_name = self.get_app_name();
```

Comment:
The app_name can be set in the URI as well ([appName option](https://www.mongodb.com/docs/manual/reference/connection-string/#mongodb-urioption-urioption.appName)).  I think the value in the URI should take precedence if set, since in the PowerBI connector that is the only way the user can modify the appName.  
We can add a check here whether it is already set in client_options, otherwise we set it to `self.get_app_name()`.